snuba/query/snql/parser.py

```diff
@@ -1317,4 +1395,26 @@ def parse_snql_query(
 
     # Validating
     _post_process(query, VALIDATORS)
+
+    if isinstance(query, LogicalQuery):  # avoid composite for now
+        if is_in_experiment(query, "", "snql.sort.query.projects", None):
+            # Experiment with sorting the query to improve cache hit rate
+            sorting_rate = state.get_config("snql.sort.query.rate", 0.0)
+            sorting_rate = 0.0 if not isinstance(sorting_rate, float) else sorting_rate
+
+            timer = Timer("query.sorting")
```

Comment:
This is going to record two metrics instead of one (you only have one mark so it will always have the same duration as the global timer value). You can avoid that by just using the `timing` method on the MetricsBackend object

Reply:
Why is this going to send two metrics? And why would the duration be the same as the global one?

Reply:
Why would this send two metrics? And why would the duration be wrong?

Reply:
Mh actually you are right it does not. Ignore